classes/blocks/class-takeactionboxout.php

```diff
@@ -168,13 +169,15 @@ public function prepare_data( $fields ): array {
 					];
 				}
 			}
+		} else {
```

Comment:
Would you mind reverting the previous condition, so that we don't need an `else`?
```php
if ( ! $query->have_posts() ) {
    return [];
}

// If page is found populate the necessary fields for the block.
// ...
```